src/app/games/lasca/lasca.component.ts

```diff
@@ -0,0 +1,330 @@
+        if (clickedSpace.isCommandedBy(opponent)) {
+            return this.cancelMove(RulesFailure.CANNOT_CHOOSE_OPPONENT_PIECE());
+        }
+        if (this.validClicks.length === 0) {
+            return this.trySelectingPiece(clickedCoord);
+        } else {
+            return this.moveClick(clickedCoord);
+        }
+    }
+    public cancelMoveAttempt(): void {
+        this.validClicks = [];
+        this.captureds = [];
+        this.updateAdaptedBoardFrom(this.getState());
+        this.rotateAdaptedBoard();
+    }
+    public async moveClick(clicked: Coord): Promise<MGPValidation> {
+        if (clicked.equals(this.validClicks[0])) {
+            this.cancelMoveAttempt();
+            return MGPValidation.SUCCESS;
+        }
+        const clickedSpace: LascaSpace = this.getState().getPieceAt(clicked);
+        const player: Player = this.getState().getCurrentPlayer();
+        if (clickedSpace.isCommandedBy(player)) {
+            this.cancelMoveAttempt();
+            return this.trySelectingPiece(clicked);
+        }
+        if (this.validClicks.length === 1) {
+            // Doing the second click, either a step or a first capture
+            const delta: Coord = this.validClicks[0].getVectorToward(clicked);
+            if (Math.abs(delta.x) === 1 && Math.abs(delta.y) === 1) {
+                // It is indeed a step
+                const step: LascaMove = LascaMove.fromStep(this.validClicks[0], clicked).get();
+                return this.chooseMove(step, this.getState());
+            }
+        }
+        // Continuing to capture
+        return this.capture(clicked);
+    }
+    private async capture(clicked: Coord): Promise<MGPValidation> {
+        const numberOfClicks: number = this.validClicks.length;
+        const lastCoord: Coord = this.validClicks[numberOfClicks - 1];
+        const delta: Coord = lastCoord.getVectorToward(clicked);
+        if (Math.abs(delta.x) === 2 && Math.abs(delta.y) === 2) {
+            const captured: Coord = new Coord(lastCoord.x + (delta.x / 2), lastCoord.y + (delta.y / 2));
+            this.captureds.push(captured);
+            this.validClicks.push(clicked);
+            const currentMove: LascaMove = LascaMove.fromCapture(this.validClicks).get();
+            if (this.validCaptures.some((capture: LascaMove) => capture.getRelation(currentMove) === 'PREFIX')) {
+                return this.applyPartialCapture();
+            } else {
+                return this.chooseMove(currentMove, this.getState());
+            }
+        } else {
+            return this.cancelMove(LascaMoveFailure.CAPTURE_STEPS_MUST_BE_DOUBLE_DIAGONAL());
+        }
+    }
+    private applyPartialCapture(): MGPValidation {
+        let partialState: LascaState = this.getState().remove(this.validClicks[0]);
+        let movingPile: LascaSpace = this.getState().getPieceAt(this.validClicks[0]);
+        for (const captured of this.captureds) {
```

Comment:
this.captureds -> this.capturedPieces, les adjectifs se pluralisent pas :p

Reply:
"alors comme ça dans les 5 autres jeux ça va mais dans celui-ci pas hein ? nameho"